src/jit/codegencommon.cpp

```diff
@@ -10853,6 +10853,139 @@ unsigned CodeGen::getFirstArgWithStackSlot()
 
 #endif // !LEGACY_BACKEND && (_TARGET_XARCH_ || _TARGET_ARM64_)
 
+void CodeGen::genSinglePush()
```

Comment:
Well, I moved it from codegenlegacy.cpp to codegencommon.cpp.

In general, I prefer to not put implementations of any kind into header files, under the assumption that a RET build will do full, global, cross-module inlining.